src/drive/lib/encryption.js

```diff
@@ -0,0 +1,35 @@
+export const generateAESKey = async ({ algorithm, keyLength } = {}) => {
+  // https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/generateKey
+  return window.crypto.subtle.generateKey(
+    {
+      name: algorithm || 'AES-GCM',
+      length: keyLength || 256
+    },
+    true, //whether the key is extractable (i.e. can be used in exportKey)
+    ['encrypt', 'decrypt']
+  )
+}
+
+export const encryptData = async (key, data, { algorithm } = {}) => {
+  const name = algorithm || 'AES-GCM'
+  // The NIST recommands 96 bits iv for AES-GCM: https://web.cs.ucdavis.edu/~rogaway/ocb/gcm.pdf
+  const iv =
+    name === 'AES-GCM'
```

Comment:
An initialization vector is needed for almost every AES-based cyphers, not only GCM (as far as I know, the only exception is the very specific AES-KW). So why having a `null` fallback for other AES crypto ?

Reply:
You could use asymetric encryption which does not require an iv